extra/script_wrappers/nanopb_generator.py.in

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+# This script is a wrapper to invoke nanopb_generator from an installed Python module.
+import sys
+import os.path
+
+# CMakeLists.txt can provide this file the path to Python module installation
+# location. It is used as a relative path. By default only system path is used.
+python_instdir = r"@PYTHON_INSTDIR@"
+cmake_bindir = r"@CMAKE_INSTALL_BINDIR@"
+cmake_install_prefix = r"@CMAKE_INSTALL_PREFIX@"
+if python_instdir[0] != '@':
+    python_instdir = os.path.join(cmake_install_prefix, python_instdir)
+    cmake_bindir = os.path.join(cmake_install_prefix, cmake_bindir)
+    relpath = os.path.relpath(python_instdir, cmake_bindir)
+    bindir = os.path.dirname(os.path.realpath(__file__))
+    libdir = os.path.abspath(os.path.join(bindir, relpath))
```

Comment:
Why don't you calculate the relative path in cmake and create two files? One with the correct name in `CMAKE_CURRENT_BINARY_DIR` and an addition file for the installation only? I would avoid a combined file because it could lead to unexpected behavior if the check is true after installation.

````
file(RELATIVE_PATH NANOPB_PYTHON_PATH "${CMAKE_CURRENT_BINARY_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}")
configure_file(
    extra/script_wrappers/nanopb_generator.py.in
    nanopb_generator.py
)
file(RELATIVE_PATH NANOPB_PYTHON_PATH "${CMAKE_INSTALL_FULL_BINDIR}" "${PYTHON_INSTDIR}")
configure_file(
    extra/script_wrappers/nanopb_generator.py.in
    nanopb_generator.py.install
)
```

Reply:
That should work also, I'm not sure which is better. I think the check is the safe way around - though I think in most cases it is not even necessary as usually Python script's own folder is already in the path.